From 56a48183b93b39f8772a9f8a2d6753c3fcffcf17 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hubert=20Figui=C3=A8re?= <hub@figuiere.net>
Date: Thu, 2 Feb 2017 13:31:46 -0500
Subject: [PATCH] Check for LLVM env to set llvm_root on initial startup

---
 tools/shared.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/shared.py b/tools/shared.py
index 2e5381b..75c9817 100644
--- a/tools/shared.py
+++ b/tools/shared.py
@@ -183,7 +183,7 @@
   config_file = '\n'.join(config_file)
   # autodetect some default paths
   config_file = config_file.replace('\'{{{ EMSCRIPTEN_ROOT }}}\'', repr(__rootpath__))
-  llvm_root = os.path.dirname(find_executable('llvm-dis') or '/usr/bin/llvm-dis')
+  llvm_root = os.environ.get('LLVM') or os.path.dirname(find_executable('llvm-dis') or '/usr/bin/llvm-dis')
   config_file = config_file.replace('\'{{{ LLVM_ROOT }}}\'', repr(llvm_root))
 
   node = find_executable('nodejs') or find_executable('node') or 'node'
-- 
2.9.3

